src/main/java/software/amazon/event/kafkaconnector/EventBridgeSinkTask.java

```diff
@@ -59,102 +71,98 @@ public void put(Collection<SinkRecord> records) {
       return;
     }
 
-    var eventBridgeWriterRecords =
-        records.stream()
-            .map(this::convertToEventBridgeWriterRecord)
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList());
+    var remainingRecords = List.copyOf(records);
 
-    // TODO: disabling batching for now until we can correctly calculate the size of the whole batch
-    // to avoid failing all entries in an EventBridge request due to exceeding 256kb size limit
-    var batches = Lists.partition(eventBridgeWriterRecords, 1);
-
-    var attempts = 0;
+    var attempts = new AtomicInteger(0);
     var start = OffsetDateTime.now(ZoneOffset.UTC);
-    while (true) {
-      attempts++;
+
+    while (!remainingRecords.isEmpty()) {
+      attempts.incrementAndGet();
       log.trace(
           "EventBridgeSinkTask putItems call started: start={} attempts={} maxRetries={}",
           start,
           attempts,
```

Comment:
question: attempts is correct use here (no get())?

Reply:
It's currently hack-ish, the reference is shared with the call to `handleFailedEntries`. Each time the writer returns with a non empty list of remaining records it's another attempt. The `AtomicInteger` does not have an `increment` method.

I'd like to refactor the retry also to a strategy to make it cleaner.